tools/synthgen/internal/synthgen/registry.go

```
@@ -0,0 +1,89 @@
+	data, err := json.Marshal(g)
+	if err != nil {
+		return g, err
+	}
+	if err := os.MkdirAll(filepath.Dir(path), 0o755); err != nil {
+		return g, err
+	}
+	f, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
+	if err != nil {
+		return g, err
+	}
+	defer f.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `f.Close` is not checked

(errcheck)